src/v1/client.rs

```diff
@@ -27,39 +27,30 @@ pub use builder::Builder;
 pub use options::Options;
 
 /// An error within the client.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum Error {
     /// An invalid request was made.
+    #[error("{0}")]
     InvalidRequest(String),
 
     /// An error when serializing or deserializing JSON.
-    SerdeJSON(serde_json::Error),
+    #[error(transparent)]
+    SerdeJSON(#[from] serde_json::Error),
 
     /// An error when serializing or deserializing JSON.
-    SerdeParams(serde_url_params::Error),
+    #[error(transparent)]
+    SerdeParams(#[from] serde_url_params::Error),
 
     /// A middleware error from `reqwest_middleware`.
     // Note: `reqwest_middleware` stores these as an [`anyhow::Error`] internally.
-    Middleware(anyhow::Error),
+    #[error(transparent)]
+    Middleware(#[from] anyhow::Error),
 
     /// An error from `reqwest`.
-    Reqwest(reqwest::Error),
-}
-
-impl std::fmt::Display for Error {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            Error::InvalidRequest(err) => write!(f, "{err}"),
-            Error::SerdeJSON(err) => write!(f, "JSON serde error: {err}"),
-            Error::SerdeParams(err) => write!(f, "serde params error: {err}"),
-            Error::Middleware(err) => write!(f, "middleware error: {err}"),
-            Error::Reqwest(err) => write!(f, "reqwest error: {err}"),
```

Comment:
This prefixes were showing up in sprocket error messages and they're not terribly useful; made them just `transparent`.